language/tools/move-disassembler/src/disassembler.rs

```diff
@@ -61,25 +63,49 @@ pub struct Disassembler<'a> {
     options: DisassemblerOptions,
     // Optional coverage map for use in displaying code coverage
     coverage_map: Option<ExecCoverageMap>,
+    /// If the code being disassembled imports multiple modules of the form (a, SameModuleName)
+    /// `module_alias` will contain an entry for each distinct a
+    /// e.g., for `use 0xA::M; use 0xB::M`, this will contain [(0xA, M) -> M, (0xB, M) -> 1M]
+    module_aliases: HashMap<ModuleId, String>,
 }
 
 impl<'a> Disassembler<'a> {
     pub fn new(source_mapper: SourceMapping<'a>, options: DisassemblerOptions) -> Self {
+        let mut module_names = HashMap::new();
+        let mut module_aliases = HashMap::new();
+        source_mapper
+            .bytecode
+            .self_id()
+            .map(|id| module_names.insert(id.name().as_str().to_string(), 0));
+        for h in source_mapper.bytecode.module_handles() {
+            let id = source_mapper.bytecode.module_id_for_handle(h);
+            let module_name = id.name().as_str().to_string();
+            module_names
```

Comment:
nit: I believe because of `Deref` coercion, you can call `to_string` directly on `Symbol`.

```suggestion
            .self_id()
            .map(|id| module_names.insert(id.name().to_string(), 0));
        for h in source_mapper.bytecode.module_handles() {
            let id = source_mapper.bytecode.module_id_for_handle(h);
            let module_name = id.name().to_string();
            module_names
```